node/pkg/reporter/reporter.go

```diff
@@ -163,6 +144,59 @@ func (r *Reporter) reportDelegated(ctx context.Context, functionString string, a
 	return r.KaiaHelper.SubmitRawTx(ctx, signedTx)
 }
 
+func processDalWsRawData(data any) (SubmissionData, error) {
+	mapData, mapDataOk := data.(map[string]interface{})
```

Comment:
I think you can make this part a bit simpler if you use json Marshal and Unmarshal

you can refer to following function and its reference in my code 
https://github.com/Bisonai/orakl/blob/0fc2d384a8228d9fbfccd1a6f7c7ee61ca022800/node/pkg/websocketfetcher/common/utils.go#L88-L101

Reply:
done!